climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+        # calc impact
+        for ind_i, haz_i in enumerate(self.hazard):
+            self._impact[ind_i].calc(self.exposure,
+                         self.vulnerability,
+                         haz_i,
+                         save_mat=True)
+            if check_plot:
+                self._impact[ind_i].plot_hexbin_eai_exposure()
+
+    def plot_imp_map(self,
+                     run_datetime=None,
+                     save_fig=True,
+                     close_fig=True,
+                     shapes_file=None):
+        """ plot a map of the impacts
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        save_fig: bool, optional
+            True default to save the figure.
+        close_fig: bool, optional
+            True default to close the figure.
+        shapes_file: str, optional
+            Points to a .shp-file with polygons in crs=epsg(21781).
+        """
+        # select hazard with run_datetime
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        # plot impact map of all runs
+        map_file_name = (self.summary_str(run_datetime) +
+                         '_impact_map' +
+                         '.jpeg')
+        map_file_name_full = Path(FORECAST_PLOT_DIR) / map_file_name
+        lead_time_str = '{:.0f}'.format(self.lead_time(run_datetime).days +
+                                        self.lead_time(run_datetime).seconds/60/60/24) # cant show '2' and '2.5'
+        title_dict = {'event_day': self.event_date.strftime('%a %d %b %Y 00-24UTC'),
+                      'run_start': (run_datetime.strftime('%d.%m.%Y %HUTC +') +
+                                    lead_time_str +
+                                    'd'),
+                      'explain_text': ('mean building damage caused by wind'),
+                      'model_text': "CLIMADA IMPACT"}
+        f, _ = self._plot_imp_map(
+                             run_datetime,
+                             title=title_dict,
+                             cbar_label=('forecasted damage per gridcell [' +
+                                         self._impact[haz_ind].unit +
+                                         ']'),
+                             polygon_file=shapes_file
+                             )
+        if save_fig:
+            f.savefig(map_file_name_full)
+        if close_fig:
+            f.clf()
+            plt.close(f)
+
+    def _plot_imp_map(self, run_datetime, title, cbar_label, polygon_file=None):
+        # select hazard with run_datetime
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        # tryout new plot with right projection
+        extend = 'neither'
+        crs_epsg = ccrs.Mercator()
```

Comment:
Why use `Mercator` instead of the CLIMADA default?

Reply:
It is `climada.util.constants.DEF_EPSG` and is equal to `4326`.

Reply:
I will use ccrs.PlateCarree() it is the most used in the code base.

Reply:
It is now more consistent and I also introduced the parameter proj everywhere so that an individual projection can be selected.